### ISSUE TITLE
Notification on desktop

### ISSUE THREAD

**Issue by at15** [dyweb/Ayi#10]
popup notification when test finished etc

see https://github.com/jolicode/JoliNotif

**Comment by at15** [comment]
the two golang notificator are all not good enough, the first one is too simple, I can just rewrite it, and it use `growlnotify` but I didn't see it on windows, nor does it include a binary file. the second one is for OSX only .... I think the php and js one worth more investigation.

**Comment by at15** [comment]
e.... visual studio community is updating .... (and it seems the update service is blocked)

**Comment by at15** [comment]
Update on how to install vs 

- VS install sucks ... it takes hours and end up with errors telling you it didn't install any extra packages except vs itself
- The install program itself is also slow ..... 

Solutions

- http://www.howtosolutions.net/2015/08/solving-installation-is-stuck-problem-in-visual-studio-community-edition/

> all they have to do is run the setup as an admin with the **/NoWeb** argument and the setup will succeed in reasonable time. The real culprit is not the setup itself, but that it tries to download and apply optional items not included in the .iso file.

And no need to mention microsoft got tons of services blocked in China, f**k, you got Azure running in China's DC, why can't you f**king use them for you f**king services. ( e.. or maybe it's the DNS services got blocked by the wall .... )

**Comment by at15** [comment]
- https://github.com/nels-o/toaster can be used by looking at code from https://github.com/jolicode/JoliNotif

**Comment by at15** [comment]
Also it's possible to use desktop notification from web interface

**Comment by xsfour** [comment]
In order to use traditional windows application, an example can be found here https://www.whitebyte.info/programming/c/how-to-make-a-notification-from-a-c-desktop-application-in-windows-10

The original post by MS is https://msdn.microsoft.com/en-us/library/windows/desktop/hh802768(v=vs.85).aspx which requires a shortcut on start menu, however, remove the related code just works ..... 

Another issue worth mentioning is that previous examples are using universal windows app, which need to have interface and use background tasks to receive event when it is minimized by user.

**Comment by xsfour** [comment]
For win32 app (which is a normal desktop application .... )

- https://blogs.msdn.microsoft.com/tiles_and_toasts/2015/10/16/quickstart-handling-toast-activations-from-win32-apps-in-windows-10/ 
- https://github.com/WindowsNotifications/desktop-toasts

**Comment by at15** [comment]
- https://github.com/Nickersoft/push.js can be used for browser based desktop nofication

**Comment by at15** [comment]
- https://github.com/sciactive/pnotify another library for browser based desktop notification

**Comment by at15** [comment]
- https://github.com/alexflint/gallium someone is trying build an electron for golang ..... though only mac is supported.
